core/src/main/scala/akka/persistence/cassandra/cleanup/Cleanup.scala

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.cassandra.cleanup
+
+import scala.collection.immutable
+import scala.concurrent.Future
+import scala.util.Failure
+import scala.util.Success
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.annotation.ApiMayChange
+import akka.event.Logging
+import akka.pattern.ask
+import akka.persistence.Persistence
+import akka.persistence.cassandra.journal.CassandraJournal
+import akka.persistence.cassandra.snapshot.CassandraSnapshotStore
+import akka.util.Timeout
+
+/**
+ * Tool for deleting all events and/or snapshots for a given list of `persistenceIds` without using persistent actors.
+ * It's important that the actors with corresponding `persistenceId` are not running
+ * at the same time as using the tool.
+ *
+ * If `neverUsePersistenceIdAgain` is `true` then the highest used sequence number is deleted and
+ * the `persistenceId` should not be used again, since it would be confusing to reuse the same sequence
+ * numbers for new events.
+ *
+ * When a list of `persistenceIds` are given they are deleted sequentially in the order
+ * of the list. It's possible to parallelize the deletes by running several cleanup operations
+ * at the same time operating on different sets of `persistenceIds`.
+ */
+@ApiMayChange
+final class Cleanup(system: ActorSystem, settings: CleanupSettings) {
```

Comment:
I don't remember the Akka version vs Cassandra but possibly ClassicActorSystemProvider to allow typed actor system as well.

Reply:
ok, can do that when porting to master, where we are 2.6 only